apps/web/src/pages/templates/components/BlueprintModal.tsx

```diff
@@ -118,7 +126,8 @@ export function BlueprintModal() {
             data-test-id="create-from-blueprint"
             onClick={() => {
               if (blueprintId) {
-                mutate(blueprintId);
+                const templatePayload = mapBlueprintToTemplateCreatePayload();
+                mutate(templatePayload);
```

Comment:
I would move this `mapBlueprintToTemplateCreatePayload` logic to the `useMutation` first argument, like:
```
useMutation(() => {
    const templatePayload =  { ...blueprint };
    templatePayload.notificationGroupId = templatePayload._notificationGroupId;
    templatePayload.blueprintId = templatePayload._id;

    return createTemplate(templatePayload);
}, {
...
});


```